engine/function/src/templates/Basic.ftl

```diff
@@ -1688,6 +1714,14 @@ public class Basic {
         return ifelse(new BooleanVectorDirect(condition), trueCase, falseCase);
     }
 
+    /**
+     * Returns an array with null values replaced with preceding non-null value
+     *
+     * @param values values.
+     * @return An array of ${pt.primitive} whose values are determined by the preceding elements.
+     *         most recently encountered non-null value where one exists
+     *         null if all preceding values were null
+     */
     public static ${pt.primitive}[] forwardFill(${pt.primitive}... values){
         if(values == null){
```

Comment:
Spacing: `if (`

So..... I've just noticed that there's a lot of these spacing things in `Basic.ftl` (not your fault). Assuming Chip agrees, maybe you can just do a global replace of `if(` to `if (` and `for(` to `for (` and `}{` to `} (` in this or a separate PR (or I can do it if that's more appropriate).

Reply:
I did this, will make re-approval more difficult but looks better, agreed.